lldb/source/Utility/ArchSpec.cpp

```diff
@@ -1439,6 +1439,10 @@ bool lldb_private::operator==(const ArchSpec &lhs, const ArchSpec &rhs) {
   return lhs.GetCore() == rhs.GetCore();
 }
 
+bool lldb_private::operator!=(const ArchSpec &lhs, const ArchSpec &rhs) {
```

Comment:
I thought this would be compiler generated if the opposite one was implemented. Is that not the case ?

Reply:
it seems not until c++20 https://en.cppreference.com/w/cpp/language/default_comparisons.html

Reply:
I'm not sure what the compiler is supposed to do, but without declaring and implementing `operator!=`, the compiler emits:

```
OptionValueArch.cpp:36:25: error: invalid operands to binary expression ('ArchSpec' and 'ArchSpec')
36 |         m_current_value != m_default_value && m_default_value.IsValid()) {
   |         ~~~~~~~~~~~~~~~ ^  ~~~~~~~~~~~~~~~
```